spring-integration-core/src/main/java/org/springframework/integration/splitter/AbstractMessageSplitter.java

```diff
@@ -247,6 +248,9 @@ protected int obtainSizeIfPossible(Iterable<?> iterable) {
 		if (iterable instanceof Collection<?> collection) {
 			return collection.size();
 		}
+		else if (JacksonPresent.isJackson3Present() && JacksonTreeNodeHelper.isNode(iterable)) {
+			return JacksonTreeNodeHelper.nodeSize(iterable);
+		}
 		else if (JacksonPresent.isJackson2Present() && JacksonNodeHelper.isNode(iterable)) {
 			return JacksonNodeHelper.nodeSize(iterable);
```

Comment:
Since this class is deprecated, we need to add a `@SuppressWarnings("removal")` to this line or the whole method.